### DISCUSSION TITLE
yield return range

### DISCUSSION THREAD

**Discussion by rromerik** [dotnet/csharplang#3949] [General] [answered]
a lot of time, when parsing a graph, we have to return all the childs elements...

   IEnumerable<string> GetNames()
   {
      yield return Name;
      foreach(var child in Childs)
      {
         foreach(var name in child.GetNames()) 
              yield return name;
      }
   }

would be nice to have a yield return range
   IEnumerable<string> GetNames()
   {
      yield return Name;
      foreach(var child in Childs)
      {
         yield return range child.GetNames();
      }
   }

**Comment by HaloFour** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/378

**Reply by rromerik** [reply]
thanks I did search before posting, didn't find it!

**Reply by HaloFour** [reply]
No worries, even when the title is more obvious it can be difficult to find existing discussions/proposals.  😄

**Comment by spydacarnage** [comment]
In the meantime, you could also (in this case, at least), use:
```csharp
IEnumerable GetNames()
{
    yield return Name;
    foreach (var name in Childs.SelectMany(c => c.GetNames())
    {
        yield return name;
    }
}
```

**Reply by leandromoh** [reply]
F# has `yield` and `yield!` (reads as yield bang) operators! this last is like your `yield return range` operator
https://stackoverflow.com/a/44947859/4854924
